net2/BroDetect.js

```diff
@@ -1082,72 +1086,24 @@ module.exports = class {
         if (flag) {
           flowspec.f = flag;
         }
-      }
+        flowspec.uids.includes(obj.uid) || flowspec.uids.push(obj.uid)
 
-      let tmpspec = {
-        ts: obj.ts, // ts stands for start timestamp
-        ets: obj.ts + obj.duration, // ets stands for end timestamp
-        sh: host, // source
-        _ts: now, // _ts is the last time updated
-        dh: dst, // dstination
-        ob: Number(obj.orig_bytes), // transfer bytes
-        rb: Number(obj.resp_bytes),
-        ct: 1, // count
-        fd: flowdir, // flow direction
-        intf: intfId, // intf id
-        tags: tags,
-        lh: lhost, // this is local ip address
-        mac: localMac, // mac address of local device
-        du: obj.duration,
-        bl: 0,
-        pf: {},
-        af: {},
-        pr: obj.proto,
-        f: flag,
-        flows: [flowDescriptor],
-        uids: [obj.uid]
-      };
+        if (obj['id.orig_p'] && !flowspec.sp.includes(obj['id.orig_p'])) {
+          flowspec.sp.push(obj['id.orig_p']);
+        }
+      }
 
-      let afobj = this.lookupAppMap(obj.uid);
+      const afobj = this.lookupAppMap(obj.uid);
       if (afobj) {
         tmpspec.af[afobj.host] = afobj;
-        let flow_afobj = flowspec.af[afobj.host];
-        if (flow_afobj) {
-          flow_afobj.rqbl += afobj.rqbl;  // request_body_len
-          flow_afobj.rsbl += afobj.rsbl;  // response_body_len
-        } else {
+        if (!flowspec.af[afobj.host]) {
           flowspec.af[afobj.host] = afobj;
-          delete afobj['host'];
-        }
-      } else {
-        flowspec._afmap[obj.uid] = obj.uid;
-        // redo some older lookup ...
-        for (let i in flowspec._afmap) {
```

Comment:
this host resolution doesn't change tmpspec, which means it doesn't change the redis object for the current flow. Same check is done on flowstash processing anyway. So remove.